rdkcentral/Dobby
# Refactor settings file and RDK platforms options

### Description
Remove platform-specific settings files from repo. Add new `-DSETTINGS_FILE` and `-DSETTINGS_APPEND` options to allow choosing a settings file in the recipe at build time.

### Test Procedure
* Build Dobby with `-DRDK_PLATFORM=DEV_VM` as normal - Dobby should use `dobby.dev_vm.json` settings file
* Build Dobby with no RDK_PLATFORM set and `-DSETTINGS_FILE` pointing to a settings file. That file will be installed in /etc/dobby.json
* Build Dobby with `-DSETTINGS_FILE` and `-DSETTINGS_APPEND` pointing to a second (smaller) settings file. The settings in the second file should be merged with the `SETTINGS_FILE` file

### Type of Change

### Requires Bitbake Recipe changes?

* Settings files should be stored in Gerrit
* -DSETTINGS_FILE and -DSETTINGS_APPEND need adding accordingly
